src/libsystemd/sd-journal/journal-file.c

```diff
@@ -878,6 +881,10 @@ int journal_file_move_to_object(JournalFile *f, ObjectType type, uint64_t offset
 
         assert(f);
 
+        /* Even if this is failed, this may clear, overwrite, or alter previously cached entries. When this
+         * is called, all objects except for one obrained by this function must be re-read and do not reuse
+         * them as is. */
```

Comment:
"Even if this function fails, it may clear, overwrite, or alter previously cached entries. After this function has been called, all objects except for one obtained by this function are invalidated and must be re-read before use."